src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -594,6 +594,10 @@ int JVM_HANDLE_XXX_SIGNAL(int sig, siginfo_t* info,
   }
 #endif
 
+  if (!signal_was_handled && sig == BREAK_SIGNAL) {
```

Comment:
I would print out a little message, since callers of SIGQUIT may expect something to happen here. Maybe "Thread dumps not yet available"?
Also, just for clarity, assert(!ReduceSignalHandling)? (but we don't do that for the other signals we omit handling either, so maybe not)

Reply:
This is in the signal handler. Is it safe to print message or do something complex?

Reply:
I would not print anything here, first because it is in the signal handler and secondly I don't think it is needed. No real thread dump usages are going to hit a VM that hasn't initialized yet IMO.

Reply:
> This is in the signal handler. Is it safe to print message or do something complex?

raw write to stdout should be fine I think.

>I would not print anything here, first because it is in the signal handler and secondly I don't think it is needed. No real thread dump usages are going to hit a VM that hasn't initialized yet IMO.

Might be nice for the user to know why his jcmd or kill -3 gets ignored though.

Reply:
because JVM_HANDLE_XXX_SIGNAL is installed in os::init_2(), it's safe to use log. I added a log entry with the tag os+init.  -Xlog:os+init=info will print out a message "ignore BREAK_SIGNAL in the initialization phase." I also update the testing example in the description section.

Reply:
Sorry for being difficult, but I would not do UL logging here. UL is heavyweight, and a shifting target too, because folks like to extend it with new capabilities. God knows what log_info really does (initialization, network traffic, accessing thread locals...) or will do in the future.

I'd keep it simple stupid and either just use write to fd=2 or fprintf(stderr). The former is safe to use in signal handling contexts, not sure about the latter. Otherwise maybe David is right and we skip the output altogether.

Reply:
Do not use UL from a signal handler! It is not safe. printf may not even be safe.

I really do not think we need to try and be this accommodating to what seems an unrealistic scenario.

Reply:
okay. I remove log_info() from signal handler. 

I don't think it's a good idea to inject a message to stdout/stderr. 1) it's inconsistent to other signals in the signal handler. 2) some java applications expect to certain outputs because they may pipe them. Injecting a message for SIGQUIT from attach client may mess them up.  I think it's fine to silence it.